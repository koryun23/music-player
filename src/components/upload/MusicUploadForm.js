import React, { useState } from "react";
import "../../css/upload/MusicUploadForm.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAdd } from "@fortawesome/free-solid-svg-icons";
import MusicUploadButton from "./MusicUploadButton";
import MusicUploadIndicator from "./MusicUploadIndicator";
import { useEffect } from "react";
export default function MusicUploadForm(props) {

    const [fileName, setFileName] = useState("");
    const [songName, setSongName] = useState("");
    const [artistName, setArtistName] = useState("");
    const [uploadIndicator, setUploadIndicator] = useState(false);
    const changeSongName = (event) => {
        setSongName(event.target.value);
    }

    const changeArtistName = (event) => {
        setArtistName(event.target.value);
    }

    const changeSelectedFile = (event) => {
        console.log(event.target.files);
        let enteredFileName = event.target.value;
        enteredFileName = enteredFileName.slice(enteredFileName.lastIndexOf("\\") + 1);
        let format = enteredFileName.slice(enteredFileName.lastIndexOf("."));
        if(format !== ".mp3" && format !== ".wav") {
            props.onError(`Cannot accept a ${format} file: Only accepting .mp3 and .wav files`);
            props.removeError()
            setSongName("");
            setArtistName("");
            setUploadIndicator(false);
            return;
        }
        setFileName(enteredFileName);
        props.onUploadSong(event);
    }

    const addSongToPlaylist = (event) => {
        event.preventDefault();
        console.log("Adding a song to the playlist");
        if(!fileName) {
            props.onError("Please upload the song");
            props.removeError();
            return;
        }
        if(!songName) {
            props.onError("Please specify the name of the song");
            props.removeError();
            return;
        }
        if(!artistName) {
            props.onError("Please specify the artist");
            props.removeError();
            return;
        }
        setTimeout(() => {
            props.onAddSongToPlaylist(event, songName, artistName, fileName);
            
            setSongName("");
            setArtistName("");
            setFileName("");
            setUploadIndicator(false);
        }, 500);
        setUploadIndicator(true);
    }

    return (
        <form className="music-upload-form">
            <div className="form form-group">
                <input type="text" 
                       name="fileName" 
                       placeholder="No File Selected" 
                       className="file-name" 
                       value={fileName}
                       readOnly />
                <small>Supported file formats: .mp3, .wav</small>

            </div>
            <div className="form form-group">
                <input type="text" 
                       name="songName" 
                       placeholder="Song Name" 
                       className="song-name" 
                       onChange={changeSongName}
                       value={songName} />
            </div>
            <div className="form form-group">
                <input type="text" 
                       name="artistName" 
                       placeholder="Artist Name" 
                       className="artist-name" 
                       onChange={changeArtistName}
                       value={artistName}/>
            </div>
            <div className="form form-group">
                <input type="file"
                       name="submit"
                       className="btn btn-submit" 
                       onChange={(event) => changeSelectedFile(event)}/>
                <MusicUploadButton onAddSongToPlaylist={addSongToPlaylist}/>
                <MusicUploadIndicator uploadIndicator={uploadIndicator}/>
            </div>
        </form>

    );
}