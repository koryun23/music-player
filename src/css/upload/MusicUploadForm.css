.music-upload-form {
    background-color: white;
    margin: auto;
    margin-top: 10px;
    width: 30%;
    height: auto;
}

.music-upload-button {
    width: 80px;
    height: 80px;
    background-color: rgb(229, 229, 229);
    color: black;
    border: 0px;
    border-radius: 50%;
    position: relative;
    right: 0;
}

.file-name {
    width: 100%;
}

.song-name {
    width: 100%;
}

.artist-name {
    width: 100%;
}

.error {
    color: rgb(255, 93, 93);
}