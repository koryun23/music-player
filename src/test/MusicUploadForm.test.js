import renderer from "react-test-renderer";
import MusicUploadForm from "../components/upload/MusicUploadForm";

it('music upload form renders correctly', () => {
    const tree = renderer.create(<MusicUploadForm />).toJSON();
    expect(tree).toMatchSnapshot();
})