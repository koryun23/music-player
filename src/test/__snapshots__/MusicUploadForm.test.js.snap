// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`music upload form renders correctly 1`] = `
<form
  className="music-upload-form"
>
  <div
    className="form form-group"
  >
    <input
      className="file-name"
      name="fileName"
      placeholder="No File Selected"
      readOnly={true}
      type="text"
      value=""
    />
    <small>
      Supported file formats: .mp3, .wav
    </small>
  </div>
  <div
    className="form form-group"
  >
    <input
      className="song-name"
      name="songName"
      onChange={[Function]}
      placeholder="Song Name"
      type="text"
      value=""
    />
  </div>
  <div
    className="form form-group"
  >
    <input
      className="artist-name"
      name="artistName"
      onChange={[Function]}
      placeholder="Artist Name"
      type="text"
      value=""
    />
  </div>
  <div
    className="form form-group"
  >
    <input
      className="btn btn-submit"
      name="submit"
      onChange={[Function]}
      type="file"
    />
    <button
      className="music-upload-button"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-lg "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</form>
`;
